src/vos/vos_aggregate.c

```diff
@@ -2156,12 +2192,22 @@ vos_agg_ev(daos_handle_t ih, vos_iter_entry_t *entry,
 	struct agg_merge_window	*mw = &agg_param->ap_window;
 	struct evt_extent	 phy_ext, lgc_ext;
 	int			 rc = 0;
+	int                      next_idx;
 
 	D_ASSERT(agg_param != NULL);
 	D_ASSERT(acts != NULL);
 	recx2ext(&entry->ie_recx, &lgc_ext);
 	recx2ext(&entry->ie_orig_recx, &phy_ext);
 
+	if (agg_param->ap_trace_count == EV_TRACE_MAX) {
+		next_idx                  = agg_param->ap_trace_start;
+		agg_param->ap_trace_start = (agg_param->ap_trace_start + 1) % EV_TRACE_MAX;
```

Comment:
ap_trace_count++ is missed?

Reply:
No, if it's EV_TRACE_MAX, it starts doing circular buffer and replacing the oldest entry.

Reply:
Right, but we could still increase the 'ap_trace_count' so that we would be able to know the exact number of entries (how many entries were truncated).

Reply:
Ah, that makes sense, fixed.